### DISCUSSION TITLE
Manual Configuration of `GnuPlot` Path on Windows

### DISCUSSION THREAD

**Discussion by RoyiAvital** [mbaz/Gaston.jl#175] [Q&A] [answered]
Is there a way to let `Gaston.jl` know the path to `GnuPlot` manually?
I guess it assumes it to be on `%PATH%` but I wonder if it can be set manually.

I am asking in the context of using `Gaston.jl` in `Plot.jl`.

**Comment by mbaz** [comment]
You're correct; it is assumed that `gnuplot` is on the path. If it's not, the easiest workaround is to modify this line, in your local package, to include the full path: https://github.com/mbaz/Gaston.jl/blob/301e3a438ede65d4ab30cb79eca01f6c49caa276/src/Gaston.jl#L76

**Reply by RoyiAvital** [reply]
Any chance to make the package aware of a system environment variable?
Somethin like `GASTON_GNU_PLOT_BIN_PATH` or something like that?

**Comment by mbaz** [comment, marked as answer]
It should be possible to do that. I'll try to include this in the upcoming v2.0 release in a few weeks.

By the way, all gnuplot Windows installers I've tried in the past provide an option to add it to the PATH. Any particular reason why that's not an option for you?

**Reply by RoyiAvital** [reply]
I prefer a portable approach where everything is set per task and not globally.
Also I saw only a ZIP file.

**Reply by RoyiAvital** [reply]
Solved by https://github.com/mbaz/Gaston.jl/issues/176.

**Comment by mbaz** [comment]
Fair enough.

There used to be some actual MSI installers that allowed you to add gnuplot to the PATH, but I haven't checked in a while.

One more thing, in my experience gnuplot/Gaston on Windows works best with the `windows` terminal. You may want to set it as default in your gnuplot config file.

**Reply by RoyiAvital** [reply]
So we also need a path for that? Where will it look for that config file on the ZIP?
I thought it would open a Figure Window (I am looking for MATLAB like experience).

By the way, are you from DSP StackExchange?

**Reply by mbaz** [reply]
No path needed; see the section on "Initialization" in gnuplot's manual (page 56 in the PDF). It also accepts commands when invoking it, e.g. `gnuplot -e "set term windows"` (see command line options in page 31).

Yep, that's me from DSP SE. I was surprised not to see your usual lion avatar here on github :)

**Reply by RoyiAvital** [reply]
Actually, yesterday I posted an answer (See [Using Least Mean Square (LMS) Filter for Beamforming on Linear Array in Julia](https://dsp.stackexchange.com/questions/81138)) with Julia code for the first time.

There was a need to plot the massive data and actually everything failed but `PlotLyJS.jl` which was a surprised (I didn't try `PyPlot.jl` but GR and the others).

I tried `Gaston.jl` which led me to this issue :-).

I really hope one day Julia will have the smooth plotting experience of MATLAB.
MATLAB's `figure` element should be a target in my opinion.
